src/Widgets/NotificationEntry.vala

```diff
@@ -231,6 +265,28 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
 
         show_all ();
 
+        button_release_event.connect (() => {
+            unowned var action_group = get_action_group (ACTION_GROUP_PREFIX);
+            if (action_group.has_action (notification.default_action)) {
+                notification.app_info.launch_action (Notification.DEFAULT_ACTION, new GLib.AppLaunchContext ());
```

Comment:
this is wrong, notifications actions aren't desktop actions.

Reply:
Hmm I took that from [the previous implementation](https://github.com/elementary/wingpanel-indicator-notifications/blame/fa7956c29251eb8db60e8053e41d772942d594cf/src/Services/Notification.vala#L149). Maybe it's relied upon by an elementary app even though it's not in the spec? Or maybe @danirabbit can give background on that?

Reply:
I’m not sure what is meant by “notifications aren’t desktop actions” but this is meant to support https://valadoc.org/gio-2.0/GLib.Notification.set_default_action.html

Reply:
> I’m not sure what is meant by “notifications aren’t desktop actions”…

a action name in the `actions` parameter of `org.freedesktop.Notifications.Notify` method aren't a "exported" action (a action in the [org.gtk.Actions](https://wiki.gnome.org/Projects/GLib/GApplication/DBusAPI#org.gtk.Actions) interface in the applications's object path), or a [desktop action](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s11.html).

`GLib.DesktopAppInfo.launch_action()` is meant to launch desktop actions only (can be abused to launch exported actions because of how GLib.Application is written, but won't work for non-GLib.Application applications).

>…but this is meant to support https://valadoc.org/gio-2.0/GLib.Notification.set_default_action.html

when GLib.Notification uses the `org.freedesktop.Notifications` interface, it will use `GLib.Action.print_detailed_name()` for buttons, and `"default"` for the default action. so we will never receive a real action that could be used with `GLib.DesktopAppInfo.launch_action()`, only false positives.

Reply:
It only launches the AppInfo now without an action. I think that's something we want or shouldn't we launch it at all? IMHO it's probably expected to bring the app to foreground when the notification is clicked on regardless of default action or no action.